3.2/root/usr/share/container-scripts/mongodb/init-petset-replset.sh

```diff
@@ -84,7 +84,12 @@ function setup_primary() {
     local config="{_id: \"${MONGODB_REPLICA_NAME}\", members: [ {_id: 0, host: \"${member_host}\"} ]}"
 
     info "Initiating MongoDB replica using: ${config}"
-    mongo admin --eval "rs.initiate(${config});${WAIT_PRIMARY_STATUS}" --quiet
+    mongo admin --eval "
+      if (db.isMaster().setName != \"${MONGODB_REPLICA_NAME}\") {
+        rs.initiate(${config});
+        ${WAIT_PRIMARY_STATUS}
+      }
+    " --quiet
```

Comment:
I tested, however in a separate VM using [`mlaunch`](https://github.com/rueckstiess/mtools/wiki/mlaunch), and a replica set can be stopped (all members) and re-started without losing data.